AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -546,4 +561,83 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
     [encoder encodeObject:_userId forKey:@"userId"];
 }
 
+- (void)addUserAttribute:(NSArray *)theValue
+                  forKey:(NSString *)theKey {
+    if (!theKey) return;
+    @synchronized(self) {
+        if (theKey != nil && theValue != nil && ![self hasUserAttributeForKey:theKey]) {
+            if (self.currentNumOfUserAttributes < MAX_NUM_OF_METRICS_AND_ATTRIBUTES) {
+                [self.userAttributes setValue:[AWSPinpointEndpointProfileUser processUserAttributeValues:theValue]
+                                       forKey:[AWSPinpointEndpointProfileUser trimKey:theKey
+                                                                              forType:@"userAttribute"]];
+            } else {
+                AWSDDLogWarn(@"Max number of userAttributes/metrics reached, dropping attribute with key: %@", theKey);
+            }
+        } else if ([self hasUserAttributeForKey:theKey]) {
+            [self.userAttributes setValue:[AWSPinpointEndpointProfileUser processUserAttributeValues:theValue]
+                                   forKey:[AWSPinpointEndpointProfileUser trimKey:theKey
+                                                                          forType:@"userAttribute"]];
+        }
+    }
+}
+
+- (NSDictionary *)allUserAttributes {
+    @synchronized(self) {
+        return [NSDictionary dictionaryWithDictionary:self.userAttributes];
+    }
+}
+
+- (BOOL)hasUserAttributeForKey:(NSString *)theKey {
+    if(!theKey) return NO;
+    @synchronized(self) {
+        if([self.userAttributes objectForKey:theKey]) {
+            return YES;
+        } else {
+            return NO;
+        }
+    }
+}
+
+- (NSArray *)userAttributeForKey:(NSString *)theKey {
+    @synchronized(self) {
+        return [self.userAttributes objectForKey:theKey];
+    }
+}
+
++ (NSArray *)processUserAttributeValues:(NSArray*) values {
+    NSMutableArray *trimmedValues = [NSMutableArray arrayWithCapacity:MAX_ENDPOINT_ATTRIBUTE_VALUES];
```

Comment:
Android does not seem to reference MAX_ENDPOINT_ATTRIBUTE_VALUES.. does android need to add this too?